### PR TITLE
Fix & improve frame matching behavior

### PR DESCRIPTION
**Pull request by C-Achard**
Fixes frames being ordered in a wrong or inconsistent way when loading data across different OSs, and improves data loading logic to be safer, while retaining previous I/O specifications. 

- [x] Fix initial bug. Description : if the top-level folder in the h5 file was incorrect, frames would be incorrectly matched or crash  
       due to improper inputs being sent to` np.vectorize` in `_remap_frame_indices`
       To reproduce, drop in a folder with an h5 that has the wrong top-level folder in its MultiIndex
- [x] Update _remap_frame_indices to correctly handle insert events (currently disabled)
- [x] Unit testing -> handle #156 first and then expand
- [x] Merge #154 first

---
This pull request introduces improvements to path handling and frame indexing throughout the codebase, ensuring path normalization is OS-agnostic and refactoring frame remapping logic for robustness and clarity. It also adds comprehensive tests for these changes and removes legacy path separator utilities.

**Path normalization and indexing improvements:**

* Refactored all usages of extracting the last N path components to use the new `misc.canonicalize_path`, which normalizes separators and works consistently across operating systems. This replaces manual slicing and ensures POSIX-style paths in metadata and indexing. (`src/napari_deeplabcut/_reader.py`, `src/napari_deeplabcut/_widgets.py`) [[1]](diffhunk://#diff-1adb77a2df0b907720bf772ed569e4addb20a0b59fd5594e91d480c64e75a02fL250-R250) [[2]](diffhunk://#diff-1adb77a2df0b907720bf772ed569e4addb20a0b59fd5594e91d480c64e75a02fL266-R266) [[3]](diffhunk://#diff-8829ceca77c22a5bc6d04552ff0d27cb02632b250c2227be4c63c810148b68e6L899-R921)
* Updated `encode_categories` to support path canonicalization and natural sorting, with additional options for sorting and returning mapping dictionaries. (`src/napari_deeplabcut/_reader.py`)

**Frame index remapping logic:**

* Completely rewrote `_remap_frame_indices` in `src/napari_deeplabcut/_widgets.py` to robustly match and remap frame indices between image layers and annotation layers, handling mismatches and partial overlaps safely. It now uses canonicalized paths at variable depths to maximize matching and avoids unsafe data deletion. (`src/napari_deeplabcut/_widgets.py`)

**Testing and utility cleanup:**

* Added extensive new tests for `canonicalize_path` and `encode_categories`, covering edge cases, OS differences, numeric categories, and empty input. Removed obsolete tests and utilities for path separator conversion (`to_os_dir_sep`). (`src/napari_deeplabcut/_tests/test_misc.py`) [[1]](diffhunk://#diff-11981a51985da17566d254dea72eb390a6a741a0a89905033f0f244750312ae3L22-R261) [[2]](diffhunk://#diff-11981a51985da17566d254dea72eb390a6a741a0a89905033f0f244750312ae3L55-L68)
* Removed import of `os` from files that no longer require it due to utility cleanup. (`src/napari_deeplabcut/_tests/test_misc.py`, `src/napari_deeplabcut/misc.py`) [[1]](diffhunk://#diff-11981a51985da17566d254dea72eb390a6a741a0a89905033f0f244750312ae3L1-R1) [[2]](diffhunk://#diff-bfc20519636742cde9e8813a5fc08506def3620056704830d3e276b6a5180a75L3)

**Widget and visualization enhancements:**

* Improved the trajectory plotting widget by setting a fixed figure size and ensuring proper canvas updates when docked. Also fixed trail visibility toggling and coloring logic for single/multi-animal data. (`src/napari_deeplabcut/_widgets.py`) [[1]](diffhunk://#diff-8829ceca77c22a5bc6d04552ff0d27cb02632b250c2227be4c63c810148b68e6R366-R373) [[2]](diffhunk://#diff-8829ceca77c22a5bc6d04552ff0d27cb02632b250c2227be4c63c810148b68e6R720) [[3]](diffhunk://#diff-8829ceca77c22a5bc6d04552ff0d27cb02632b250c2227be4c63c810148b68e6L821-L843)

### CODE REVIEW

**Comment by C-Achard** [comment]
@deruyter92 If you have time could you look at this fix real quick ? I feel that one would  be good to merge soonish given what it fixes

File: src/napari_deeplabcut/misc.py [hunk 1]
```diff
@@ -9,6 +9,59 @@
...
+    """
+    try:
+        s = str(p)
+    except Exception as e:
+        logger.debug("Failed to stringify path %r (%s).", p, type(e).__name__, exc_info=True)
+        return ""
+
+    s = s.replace("\\", "/")
+    s = s.rstrip("/")
+    parts = [part for part in s.split("/") if part and part != "."]
```
Comment (deruyter92) [inline src/napari_deeplabcut/misc.py]: Good that you handle ".", maybe also add ".." ?

File: src/napari_deeplabcut/_tests/test_misc.py [hunk 2]
```diff
@@ -19,12 +19,244 @@ def test_unsorted_unique_string():
...
+    # Always normalize separators to POSIX style
+    p = r"a\b\c\file.png"
+    out = misc.canonicalize_path(p, n=3)
+    assert "\\" not in out
+
+    # Compute expected using the same logic as canonicalize_path
+    s = p.replace("\\", "/").rstrip("/")
+    parts = [part for part in s.split("/") if part and part != "."]
+    expected = "/".join(parts[-3:]) if parts else ""
+    assert out == expected
```
Comment (deruyter92) [inline src/napari_deeplabcut/_tests/test_misc.py]: To avoid circularity, maybe better to parametrize the exact input and output strings, instead of applying the same logic inside the test?

(Maybe also parameterize the test above test_canonicalize_path, makes it easy to add eventual edge cases)

**Review by deruyter92 — COMMENTED** [review]
Looks like a good fix, and happy with the improvements. Great that you add these tests. As well. Just minor comments.  Feel free to merge after addressing/disregarding!

Note: your changes look good, but some of the existing code may also need some refactoring later as well (e.g. some of the padas indexing and merging). But for now I'm favor of not touching the rest if it is working fine.

File: src/napari_deeplabcut/_tests/test_misc.py (hunk 2 above)
Comment (C-Achard) [reply src/napari_deeplabcut/_tests/test_misc.py]: Fixed, thanks a lot !

File: src/napari_deeplabcut/misc.py (hunk 1 above)
Comment (C-Achard) [reply src/napari_deeplabcut/misc.py]: Improved this a bit further

**Comment by C-Achard** [comment]
> Note: your changes look good, but some of the existing code may also need some refactoring later as well (e.g. some of the padas indexing and merging). But for now I'm favor of not touching the rest if it is working fine.

Indeed, we will need to coordinate on this. I think a nice way would be for me to write napari helpers (to_napari) etc in the main DLC codebase, so we can import them and test them from there and keep the plugin minimal.

But this all depends on how we handle the migration so probably for later. Tagging https://github.com/DeepLabCut/DeepLabCut/pull/3171 for future reference.

**Review by deruyter92 — APPROVED** [review]
